src/core/exception.d

```diff
@@ -416,12 +427,48 @@ deprecated void setAssertHandler( AssertHandler h ) @trusted nothrow @nogc
     assertHandler = h;
 }
 
+/*
+Returns a perThreadInstance object of class type `C`, i.e. repeated calls to
+`perThreadInstance!C` within a given thread return the same instance. For that reason,
+no constructor parameters are accepted; `C` must allow default construction.
+
+Different threads own different instances of `C`.
+
+The destructor of the perThreadInstance object is called during application's exit.
+*/
+private C perThreadInstance(C)() if (is(C == class) && is(typeof(new C)))
+{
+    static size_t[1 + (__traits(classInstanceSize, C) - 1) / size_t.sizeof] b;
+    static bool initialized = false;
+    auto result = () @trusted { return cast(C) b.ptr; }();
+    if (!initialized)
+    {
+        assert(typeid(C).initializer.length == b.length * size_t.sizeof);
+        import core.stdc.string;
+        () @trusted
+        {
+            memcpy(b.ptr, typeid(C).initializer.ptr, b.length);
+        }();
+        result.__ctor;
+        import core.internal.traits;
+        static if (hasElaborateDestructor!C)
```

Comment:
According to a quick test:
```D
import std.traits, core.exception;
static assert(hasElaborateDestructor!AssertError);

// Error: static assert  (hasElaborateDestructor!(AssertError)) is false
```

So it would be nice to have this line covered. Otherwise, LGTM.